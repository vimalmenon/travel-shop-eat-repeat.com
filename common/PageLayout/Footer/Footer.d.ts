import { IMetaData } from '@types';

export interface IFooterProps {
  navigation: IMetaData[];
}
