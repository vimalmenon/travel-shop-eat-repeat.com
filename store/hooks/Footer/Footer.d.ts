export type UseFooterType = () => {
  versionNumber: string;
  appName: string;
};
