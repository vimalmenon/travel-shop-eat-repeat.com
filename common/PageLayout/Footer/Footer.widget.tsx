import { useFooterHelper } from '@store';

export const Footer: React.FC = () => {
  const { versionNumber } = useFooterHelper();
  return (
    <footer className="flex text-sm py-1 px-2 justify-between">
      <span>This is footer</span>
      <span>{versionNumber}</span>
    </footer>
  );
};
