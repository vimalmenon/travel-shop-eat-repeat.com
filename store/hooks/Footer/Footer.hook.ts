import { useSelector } from 'react-redux';

import { AppState } from '../../app/app';
import { RootState } from '../../store';
import { UseFooterType } from './Footer';

export const useFooterHelper: UseFooterType = () => {
  const { version, mode } = useSelector<RootState, AppState>((state) => state.app);
  return {
    versionNumber: `${version}-${mode}`,
  };
};
