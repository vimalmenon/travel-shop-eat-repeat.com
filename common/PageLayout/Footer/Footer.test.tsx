import { useFooterHelper } from '@store';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Footer } from './Footer.widget';

jest.mock('@store');

describe('common > <PageBase> <Header> <Footer> Component', () => {
  describe('Snapshot Testing', () => {
    it('<Footer>', () => {
      const mock = useFooterHelper as jest.Mock;
      mock.mockImplementation(() => ({
        versionNumber: '0.0.0-D',
      }));
      const wrapper = shallow(<Footer />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
