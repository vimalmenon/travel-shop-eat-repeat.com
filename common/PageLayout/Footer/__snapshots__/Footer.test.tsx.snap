// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common > <PageBase> <Header> <Footer> Component Snapshot Testing <Footer> 1`] = `
<footer
  className="flex text-sm py-1 px-2 justify-between"
>
  <span>
    This is footer
  </span>
  <span>
    0.0.0-D
  </span>
</footer>
`;
